domain/secretbackend/service/service.go

```diff
@@ -700,9 +700,53 @@ func (s *Service) RotateBackendToken(ctx context.Context, backendID string) erro
 	return errors.Trace(err)
 }
 
-// SetModelSecretBackend sets the secret backend for the given model UUID.
+// GetModelSecretBackend returns the secret backend name for the given model UUID, returning an error
```

Comment:
I would like to see one big change with this service.

Can we introduce a new service like:
```go
type ModelService struct {
    Service
    modelID coremodel.UUID
}
```


Can we then re-implement Get and Set as proxies on to the `m.Service` using the stored model id. This way the secret backend service can be scoped off to one model and then the scoped model facade doesn't have the power to change a secret backend for an unrelated model and it follows our schematics of facades and services scoped to the model they work on.

It also allows the secret backend service to be exposed on the ModelServiceFactory and not the ControllerServiceFactory.

Reply:
I have to disagree with this approach. 
Wrapping a controller service into a brand new service and registering to the service factory for just exposing two new methods is just overkill. 
Just like what I did before, ideally, these two methods should be moved to `modelconfig`. But by doing this, we will have to expose too many shared helper functions in the secretbacken package, so we decided to keep them inside the current package.